nautobot/circuits/views.py

```diff
@@ -3,277 +3,282 @@
+            circuits_table.columns.hide("provider")
 
-        circuits_table = tables.CircuitTable(circuits)
-        circuits_table.columns.hide("termination_a")
-        circuits_table.columns.hide("termination_z")
+            paginate = {
+                "paginator_class": EnhancedPaginator,
+                "per_page": get_paginate_count(request),
+            }
+            RequestConfig(request, paginate).configure(circuits_table)
 
-        paginate = {"paginator_class": EnhancedPaginator, "per_page": get_paginate_count(request)}
-        RequestConfig(request, paginate).configure(circuits_table)
+            return {
+                "circuits_table": circuits_table,
+            }
+        elif view_type == "list":
+            return {}
+        elif view_type == "bulk_edit":
+            return {}
+        else:
+            return {}
 
-        return {
-            "circuits_table": circuits_table,
-        }
 
+class ProviderViewSetRouter(NautobotRouter):
+    viewset = ProviderViewSet
 
-class ProviderNetworkEditView(generic.ObjectEditView):
-    queryset = ProviderNetwork.objects.all()
-    model_form = forms.ProviderNetworkForm
+    def __init__(self, prefix, basename):
+        super().__init__(prefix, basename)
 
 
-class ProviderNetworkDeleteView(generic.ObjectDeleteView):
-    queryset = ProviderNetwork.objects.all()
+class ProviderNetworkViewSet(NautobotViewSet):
+    model = ProviderNetwork
+    table = tables.ProviderNetworkTable
+    form = forms.ProviderNetworkForm
+    filterset_form = forms.ProviderNetworkFilterForm
+    filterset = filters.ProviderNetworkFilterSet
+    import_form = forms.ProviderNetworkCSVForm
+    bulk_edit_form = forms.ProviderNetworkBulkEditForm
+    lookup_field = "slug"
+
+    def get_extra_context(self, request, view_type, instance):
+        if view_type == "detail":
+            circuits = (
+                Circuit.objects.restrict(request.user, "view")
+                .filter(Q(termination_a__provider_network=instance.pk) | Q(termination_z__provider_network=instance.pk))
+                .prefetch_related("type", "tenant", "terminations__site")
+            )
 
+            circuits_table = tables.CircuitTable(circuits)
+            circuits_table.columns.hide("termination_a")
+            circuits_table.columns.hide("termination_z")
 
-class ProviderNetworkBulkImportView(generic.BulkImportView):
-    queryset = ProviderNetwork.objects.all()
-    model_form = forms.ProviderNetworkCSVForm
-    table = tables.ProviderNetworkTable
+            paginate = {"paginator_class": EnhancedPaginator, "per_page": get_paginate_count(request)}
+            RequestConfig(request, paginate).configure(circuits_table)
 
+            return {
+                "circuits_table": circuits_table,
+            }
+        elif view_type == "list":
+            return {}
+        elif view_type == "bulk_edit":
+            return {}
+        else:
+            return {}
```

Comment:
Why is this pattern necessary? Is there any harm in just always returning the extra context?